### ISSUE TITLE
Rename of org.gradle.api.provider.PropertyState to org.gradle.api.provider.Property caused a loss in DSL extension function

### ISSUE THREAD

**Issue by sic-bre** [gradle/kotlin-dsl#574]
org.gradle.api.provider.PropertyState was renamed to org.gradle.api.provider.Property but PropertyStateExtensions.kt has no equivalent to PropertyState.kt to provide these two helpers:

package org.gradle.kotlin.dsl
import org.gradle.api.provider.Property
import kotlin.reflect.KProperty
operator fun <T> Property<T>.getValue(receiver: Any?, property: KProperty<*>): T = get()
operator fun <T> Property<T>.setValue(receiver: Any?, property: KProperty<*>, value: T) = set(value)

### Expected Behavior
Gradle 4.3 (final) should have a PropertyStateExtensions.kt with the above helpers providing equivalent functionally to ProperyState for the renamed Property class.

### Current Behavior
For projects that need this accessor I have to separately  provide my own helpers.

### Context
Every release of gradle-kotlin-dsl has had breaking changes of the incubating ProperyState/ProviderState or Property/Provider API braking binary compatibility and requiring code changes and recompiling of dependent plugins.  Hopefully this will be the last API change that brakes the simplest use cases so we can make Gradle 4.3 our stable minimum version.

### Steps to Reproduce (for bugs)
Convert any code using Kotin extension functions for PropertyState to Property and it will stop working.

### Your Environment
Converting a build from Gradle 4.1 to Gradle 4.3-rc-3.

**Comment by bamboo** [comment]
Closing as delivered (46e5e6b73dcfa15b43274556f74676d821b9e70a).

The fix will be part of Gradle 4.3 RC4.

Thanks for report, @sic-bre!

**Comment by sic-bre** [comment]
Validated that the new api as delivered in gradle-4.3-rc-4 is working in my use cases.  Not backwards binary compatible so I will have to set my corporate minimum version to gradle-4.3 when it releases.  It would be useful to add comment in the final release notes on what the expected api looks like.  Here is an example configuration helper function using the APIs that I am using on my Extension classes to simplify setting of lazy configuration values from Gradle properties with lazy defaults:

```
package com.standard.gradle.plugins.sample

import org.gradle.api.Project
import org.gradle.api.provider.Property
import org.gradle.kotlin.dsl.*
import java.util.*

open class Defaulting(val project: Project) {
    protected inline fun <reified T> defaulting(name: String, crossinline default: () -> T): Property<T> {
        val prop = project.objects.property<T>()
        val provider = project.provider<T> {
            val defaultValue = default()
            val falseRegex = Regex("^(?i)false|0|$") // Match groovy conventions for FaLSe, 0 and ""
            val property = project.findProperty(name)
            val value: Any = when (defaultValue) {
                is String -> property?.toString() ?: defaultValue
                is Boolean -> !falseRegex.matches(property?.toString() ?: defaultValue.toString())
                is Int -> property?.toString()?.toIntOrNull() ?: defaultValue
                is Short -> property?.toString()?.toShortOrNull() ?: defaultValue
                is Long -> property?.toString()?.toLongOrNull() ?: defaultValue
                is Float -> property?.toString()?.toFloatOrNull() ?: defaultValue
                is Double -> property?.toString()?.toDoubleOrNull() ?: defaultValue
                is Byte -> property?.toString()?.toByteOrNull() ?: defaultValue
                is Regex -> property?.toString()?.toRegex() ?: defaultValue
                else -> throw UnknownFormatConversionException("Unknown format for PropertyState of type")
            }
            value as T
        }
        prop.set(provider)
        return prop
    }
}
```

And here is an example usage:

```
package com.standard.gradle.plugins.sample

import org.gradle.api.Project
import org.gradle.kotlin.dsl.*

open class SamplePluginExtension(project: Project): Defaulting(project) {
    val reportName by defaulting("sample.reportName") { project.name.toString() }
    val tag by defaulting("sample.tag") { "sample" }
}
```
